gateway/src/lib.rs

```diff
@@ -283,6 +285,24 @@ impl Server {
         Ok(())
     }
 
+    /// The rack_id will be set on a refresh of the SMF property when the sled
+    /// agent starts.
+    pub fn set_rack_id(&self, rack_id: Option<Uuid>) {
+        if let Some(rack_id) = rack_id {
+            let val = self.apictx.rack_id.get_or_init(|| rack_id);
+            if *val != rack_id {
+                error!(
+                    self.apictx.log,
+                    "Rack ID changed from {} to {}", val, rack_id
+                );
+            } else {
+                info!(self.apictx.log, "Set rack_id: {rack_id} for MGS");
```

Comment:
Tiny nit - this log is just for MGS, so I think we can drop that bit. I might also shove the rack ID into a slog key/value? I haven't been consistent with this but I feel like it's a little cleaner; something like

```suggestion
                info!(self.apictx.log, "Set rack_id"; "rack_id" => %rack_id);
```

Reply:
Fixed.